chore: submodule noir

charlielye opened the pull request:
Submodules are a pain.
But at least they're standardised pain that AI can help you with.

ludamad commented:
Ah really. People are still committing submodule changes spuriously all the time... I'm not in favour of this, but won't fight it

EDIT: I have been convinced, if we have a bit of safety here. AI does help but I'd love a commit hook warning.

ludamad commented on .github/workflows/pull-noir.yml:
```diff
@@ -13,93 +13,105 @@ on:
+            LATEST_NIGHTLY="${{ github.event.inputs.ref }}"
           else
-            LATEST_NOIR_NIGHTLY=$(noir/scripts/sync.sh latest-nightly)
+            # Fetch all tags and get latest nightly
```
lgtm, but were you able to test it? if not perhaps collapsing it to a single testable script is a good idea

ludamad reviewed (approved):
approved with comment

charlielye committed:
chore: submodule noir

Submodules are a pain.
But at least they're standardised pain that AI can help you with.
